batch-stark/src/prover.rs

```diff
@@ -119,20 +142,30 @@ where
             ext_trace_domains[i].create_disjoint_domain(1 << (log_ext_degrees[i] + lqd));
 
         // Count constraints to size alpha powers packing.
-        let constraint_cnt = get_symbolic_constraints(airs[i], 0, pub_vals[i].len()).len();
+        let constraint_cnt =
+            get_symbolic_constraints(airs[i], preprocessed_widths[i], pub_vals[i].len()).len();
 
         // Get evaluations on quotient domain from the main commitment.
         let trace_on_quotient_domain =
             pcs.get_evaluations_on_domain(&main_data, i, quotient_domain);
 
+        // Get preprocessed evaluations if this instance has preprocessed columns.
+        let preprocessed_on_quotient_domain = common
+            .preprocessed
+            .as_ref()
+            .and_then(|g| g.instances[i].as_ref().map(|meta| (g, meta)))
+            .map(|(g, meta)| {
+                pcs.get_evaluations_on_domain(&g.prover_data, meta.matrix_index, quotient_domain)
```

Comment:
Does this mean you're evaluating the preprocessed polynomials during proving? Could this be done ahead of time, so that this code only reads/deserializes them?